### PR TITLE
Add example code for dialog filter and callback

### PR DESCRIPTION
**Pull request by Semphriss**
Replaces https://github.com/libsdl-org/SDL/pull/9729, the same description applies.

This is my first PR here, please let me know if I got something wrong!

### CODE REVIEW

File: SDL3/SDL_DialogFileCallback.md [hunk 1]
```diff
@@ -31,6 +31,40 @@ more than the size of the list (therefore the index of the terminating NULL
 entry) if no filter was selected, or -1 if the platform or method doesn't
 support fetching the selected filter.
 
+## Code Examples
+
+```c
+static void SDLCALL callback(void* userdata, const char* const* files, int filter) {
```
Comment (slouken) [inline SDL3/SDL_DialogFileCallback.md:37]:
```suggestion
static void SDLCALL callback(void* userdata, const char* const* files, int filter)
{
```

File: SDL3/SDL_DialogFileFilter.md [hunk 2]
```diff
@@ -27,6 +27,20 @@ document").
...
 single asterisk ("*"), which serves as an "All files" filter.
 
+## Code Examples
+
+This structure is most often used as a **NULL-terminated** array:
+
+```c
+const SDL_DialogFileFilter filters[] = {
+    { "PNG images",  "png" },
+    { "JPEG images", "jgp;jpeg" },
```
Comment (zturtleman) [inline SDL3/SDL_DialogFileFilter.md:37]: There seems to be a typo of jpg in "jgp;jpeg".

**Comment by icculus** [comment]
This looks good. I'm adding this and then fixing the syntax error.

**Comment by icculus** [comment]
Okay, make sure I didn't screw this up in 3a5ce94d54dba2268d86c2257fd73d9b2ec1b366.

(Outside of this specific example: should this callback send in the filter array too, so we don't have to have a global to figure out what the `filter` argument is referring to? Let me know if I should open an issue about it.)

**Comment by Semphriss** [comment]
I reviewed the commit and I didn't find anything wrong.

One thing though, there are still `\n`'s with SDL_Log in `test/testdialog.c`, should they be removed too?

**Comment by slouken** [comment]
SDL_Log messages should be terminated with "\n"

**Pull request merged**